response_operations_ui/common/date_restriction_generator.py

```diff
@@ -8,13 +9,22 @@ def get_date_restriction_text(tag, events):
         "exercise_end": [f"Must be after Return by {_get_event_date_string('return_by', events)}"],
         "reminder": [f"Must be after Go Live {_get_event_date_string('go_live', events)}",
                      f"Must be before Exercise end {_get_event_date_string('exercise_end', events)}"],
-        "reminder2": [f"Must be after Go Live {_get_event_date_string('go_live', events)}",
+        "reminder2": [f"Must be after First Reminder {_get_event_date_string('reminder', events)}",
                       f"Must be before Exercise end {_get_event_date_string('exercise_end', events)}"],
-        "reminder3": [f"Must be after Go Live {_get_event_date_string('go_live', events)}",
+        "reminder3": [f"Must be after Second Reminder {_get_event_date_string('reminder2', events)}",
                       f"Must be before Exercise end {_get_event_date_string('exercise_end', events)}"],
         "ref_period_start": [f"Must be before Reference Period end {_get_event_date_string('ref_period_end', events)}"],
         "ref_period_end": [f"Must be after Reference Period start {_get_event_date_string('ref_period_start', events)}"]
-    }.get(tag)
+    }
+
+    if _get_event_date_string('reminder2', events):
```

Comment:
Could you comment what this bit is doing?  I still don't really get it.

Reply:
It changes the text based on the events which have been created. eg if there is no reminder 3, on the reminder 2 update form it will say it needs to be before the return by, but if it exists it will say it must be before reminder 3.